api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
@coreyhulen I don't think we do any validation on `c.TeamId`, it looks to me like we just set it from the URL parameter and that's it
https://github.com/mattermost/platform/blob/master/api/context.go#L115

@cpanato If Corey agrees, it might be better to add the validation into context.go so that this occurs for all APIs that require a team ID. I think the check should be something like:
```
if c.Session.GetTeamByTeamId(c.TeamId) == nil {
	if HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
		// check if team exists here
	}
}
```